src/app/shared/services/courses/courses.service.ts

```diff
@@ -1,62 +1,121 @@
 import { Injectable } from '@angular/core';
-import { Course } from '../../course.model';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { Course } from '../../models/course.model';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CoursesService {
-  public coursesList: Course[] = [];
+  private COURSES_URL = 'http://localhost:3004/courses';
+  public coursesList$ = new BehaviorSubject<Course[]>([]);
+  private httpOptions = {
+    headers: new HttpHeaders({
+      'Content-Type': 'application/json'
+    })
+  };
+  private courseOnPageAmount = 3;
+  private searchingParams = { start: '0', count: `${this.courseOnPageAmount}` };
+  private isFiltered = false;
+  private searchString: string;
+
+  constructor(private http: HttpClient) {
+    this.httpRequestCourses();
+  }
 
-  getCourses(): Course[] {
-    if (this.coursesList.length === 0) {
-      return (this.coursesList = this.createCoursesList());
+  loadMoreCourses() {
```

Comment:
see comments below, try to do smth like this:
`loadMoreCourses(): Observable<Course[]> { return this.http.get<Course[]>(.....)}`